src/utils/data.js

```diff
@@ -138,7 +139,7 @@ export function clone(jsonObject) {
 }
 
 export async function fetchIndexPathsData({ currentLocale }) {
-  const locale = currentLocale === defaultLocale ? '' : currentLocale;
-  const path = new URL(`${pathJoin([baseUrl, 'index/', locale, 'index.json'])}`, window.location.href);
+  const slug = currentLocale === defaultLocale ? '' : getSlug(currentLocale);
```

Comment:
I don't think you would need any changes to this file if you were to make a minor change in `NavigatorDataProvider`—instead of passing it the value of `$i18n.locale` as a prop, it should simply pass `$route.params.locale` to this function, which should always be the URL slug so no conversion would be needed here.